source/blog/_posts/2018-05-30-pair-programming.md

```diff
@@ -0,0 +1,135 @@
+---
+title: An anecdote about pair programming
+tags:
+- Agile Development
+- Extreme programming
+- Development techniques
+
+categories: 
+- dev
+
+authors: [cr]
+github_link: /blog/_posts/2018-05-30-pair-programming.md
+
+---
+
+When we decided to rebuild a central piece of our Merchant Integration Architecture, I was assigned to write a new Component, that handles asynchronous function calls through event queueing. 
+So from the get go it was assumed that this was quite a large task. After having completed a rough prototype to interact with our queue service, I hit a problem with deciding how proceed.
+There were many paths that needed to be worked on: error handling, at least once execution, backing up the queue to our database, et cetera.
+So I asked for ideas during our daily stand-up meeting, our team lead suggested I team up with a colleague who's out of work to try [pair programming](http://www.extremeprogramming.org/rules/pair.html). 
+
+## Parallel Thinking
+At first we both had doubts about the fruitfulness of coding together and breathing down each others necks. But we quickly realized that while the typing speed remained the same the thought process was streamlined.
```

Comment:
"But we quickly realized that **while** the typing speed remained the same the thought process was streamlined." -> "But we quickly realized that the typing speed remained the same **while** the thought process was streamlined."